### ISSUE TITLE
[subplat] Currency error message displayed when trying to update card information 

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/fxa#10904]
**Affected versions**:

* All;

**Affected FxA versions**:

* 1.218.4;

**Prerequisites**:

* Be in a Mozilla VPN supported country (US);

**Steps to reproduce**:
1. Subscribe to Mozilla VPN monthly plan using Stripe and a US card (e.g. 4242424242424242);
2. Go to Subscription Management page for the same account;
3. Click on the “Change” button to update payment details (e.g. 4000056655665556);
4. Click on the “Update” button;
5. Observe the behaviour;

**Expected result**:

* The payment details are successfully updated. 

**Actual result**:

* “The currency of this subscription is not valid for the country associated with your payment” error message is displayed in the Payment section. The card number cannot be updated.

**Notes**:

* This is a major issue because the user is not able to update his payment information;
* Reproducible if using the same card number that was used for making the subscription;
* Reproducible for new and old users;
* Reproducible regardless of the currency;
* See the attached screenshot.

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/FXA-4171)
┆Attachments: <a href="https://mozilla-hub.atlassian.net/secure/attachment/17413/Screenshot+2021-11-02+at+17.47.12.png">Screenshot 2021-11-02 at 17.47.12.png</a>

**Comment by data-sync-user** [comment]
➤ Anna Sobiepanek commented:

Was able to reproduce on staging. Looks like we are getting a 400 from https://api-accounts.stage.mozaws.net/v1/oauth/subscriptions/paymentmethod/default ( https://api-accounts.stage.mozaws.net/v1/oauth/subscriptions/paymentmethod/default )  Request sends paymentMethodId: "pm_1Js6VQKb9q6OnNsLbXmursAm" Response is: 

{"code":400,"errno":130,"error":"Bad Request","message":"Funding source country does not match plan currency.","info":"https://github.com/mozilla/fxa/blob/main/packages/fxa-auth-server/docs/api.md#response-format","currency":null,"country":"US"}

**Comment by data-sync-user** [comment]
➤ Madalin Cotetiu commented:

Verified this as fixed on Train 1.218.6 - Staging. The payment details are successfully updated (for new users and old users).

We noticed that for the accounts with the subscription made on 11/04/2021, the card number cannot be updated, regardless of the country. This is not happening for the accounts with older or newer subscriptions.

Is this something to be concerned about?

Thank you.
